mitodl/odl-video-service
# Subtitles

#### What are the relevant tickets?
Fixes #216 
Fixes #217 
Fixes #218 
Fixes #219 
Fixes #220 
Fixes #221 

#### What's this PR do?
- Adds a VideoSubtitle model
- Adds support for uploading, downloading, deleting subtitle files
- Adds support for displaying subtitles w/VideoJS

#### How should this be manually tested?
- Add `VIDEO_S3_SUBTITLE_BUCKET=odl-video-service-subtitles-ci` to your .env file
- A video and subtitle file are available for download here: http://www.storiesinflight.com/js_videosub/
  - The subtitle is in SRT format, I converted it to VTT (in comments below, zipped).
- Upload the video
- Go to the video detail page
- Click on the 'Add subtitle' button
- Select the subtitle file. You should only be able to select `.vtt` files
- The subtitle should now appear in the list.
- The subtitle selector button should appear in the VideoJS controls. "English' should be one of the options.
- Selecting 'English' should display the subtitles (note: this will not work until cross-origin headers are allowed to come through on the new subtitle bucket)
- Download the subtitle
- Reload the page, the subtitle selector button should be present.
- Delete the subtitle.  The subtitle row should disappear from the card, and the subtitle selector button should disappear from VideoJS.
